File: frontend/src/lib/schemas/address.ts
import { z } from 'zod';

export const addressSchema = z.object({
	line1: z.string(),
	line2: z.string().optional(),
	city: z.string(),
	stateOrProvince: z.string().optional(),
	zip: z.string(), // rename this to postalCode
	country: z.string().min(2)
});

File: backend/internal/api/address.go
package api

type Address struct {
	Line1 string `json:"line1" validate:"required"`
	Line2 string `json:"line2" validate:"omitempty"`
	City string `json:"city" validate:"required"`
	StateOrProvince string `json:"stateOrProvince" validate:"required"`
	Zip string `json:"zip" validate:"required"`
	Country string `json:"country" validate:"required"`
}